pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -12340,6 +12340,1468 @@ spec:
+  childReferences:
+  - apiVersion: tekton.dev/v1
+    kind: TaskRun
+    name: pr-matrix-emitting-results-0
+    pipelineTaskName: matrix-emitting-results
+  - apiVersion: tekton.dev/v1
+    kind: TaskRun
+    name: pr-matrix-emitting-results-1
+    pipelineTaskName: matrix-emitting-results
+  - apiVersion: tekton.dev/v1
+    kind: TaskRun
+    name: pr-matrix-emitting-results-2
+    pipelineTaskName: matrix-emitting-results
+  - apiVersion: tekton.dev/v1
+    kind: TaskRun
+    name: pr-matrixed-echo-length
+    pipelineTaskName: matrixed-echo-length
+  - apiVersion: tekton.dev/v1
+    kind: TaskRun
+    name: pr-matrixed-echo-results-length
+    pipelineTaskName: matrixed-echo-results-length
+`),
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			pr := parse.MustParseV1PipelineRun(t, fmt.Sprintf(`
+metadata:
+  name: pr
+  namespace: foo
+spec:
+  taskRunTemplate:
+    serviceAccountName: test-sa
+  pipelineRef:
+    name: %s
+`, tt.name))
+			d := test.Data{
+				PipelineRuns: []*v1.PipelineRun{pr},
+				Pipelines:    []*v1.Pipeline{tt.p},
+				Tasks:        []*v1.Task{task1, task2, task3, task4, taskwithresults},
+				TaskRuns:     taskRuns,
+				ConfigMaps:   cms,
+			}
+			prt := newPipelineRunTest(t, d)
+			defer prt.Cancel()
+			_, clients := prt.reconcileRun("foo", "pr", []string{}, false)
+			taskRuns, err := clients.Pipeline.TektonV1().TaskRuns("foo").List(prt.TestAssets.Ctx, metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("tekton.dev/pipelineRun=pr,tekton.dev/pipeline=%s", tt.name),
+				Limit:         1,
+			})
+			if err != nil {
+				t.Fatalf("Failure to list TaskRun's %s", err)
+			}
+			if len(taskRuns.Items) != len(tt.expectedTaskRuns) {
+				t.Fatalf("Expected %d TaskRuns got %d", len(tt.expectedTaskRuns), len(taskRuns.Items))
+			}
+			for i := range taskRuns.Items {
+				expectedTaskRun := tt.expectedTaskRuns[i]
+				expectedTaskRun.Labels["tekton.dev/pipeline"] = tt.name
+				expectedTaskRun.Labels["tekton.dev/memberOf"] = tt.memberOf
+				if d := cmp.Diff(expectedTaskRun, &taskRuns.Items[i], ignoreResourceVersion, ignoreTypeMeta); d != "" {
+					t.Errorf("expected to see TaskRun %v created. Diff %s", tt.expectedTaskRuns[i].Name, diff.PrintWantGot(d))
+				}
+			}
+			pipelineRun, err := clients.Pipeline.TektonV1().PipelineRuns("foo").Get(prt.TestAssets.Ctx, "pr", metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("Got an error getting reconciled run out of fake client: %s", err)
+			}
+			if d := cmp.Diff(tt.expectedPipelineRun, pipelineRun, ignoreResourceVersion, ignoreTypeMeta, ignoreLastTransitionTime, ignoreStartTime, ignoreFinallyStartTime, ignoreProvenance, cmpopts.EquateEmpty(), cmpopts.SortSlices(lessChildReferences)); d != "" {
+				t.Errorf("expected PipelineRun was not created. Diff %s", diff.PrintWantGot(d))
+			}
+		})
+	}
+}
+
+func TestReconciler_PipelineTaskMatrixConsumingResults2(t *testing.T) {
```

Comment:
what's the difference between this `nc TestReconciler_PipelineTaskMatrixConsumingResults2` and the above `nc TestReconciler_PipelineTaskMatrixConsumingResults`? We shouldn't use this naming

Reply:
Will rename. The first one is testing matrix with include params. The second has regular matrix params.